### PR TITLE
Add query to log in getFromDBByCrit

### PR DESCRIPTION
**Pull request by nenes82**
Add the query in the log to help find faster the problem

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

File: inc/commondbtm.class.php [hunk 1]
```diff
@@ -347,7 +347,7 @@ public function getFromDBByCrit(array $crit) {
       } else if (count($iter) > 1) {
          Toolbox::logWarning(
             sprintf(
-               'getFromDBByCrit expects to get one result, %1$s found!',
+               'getFromDBByCrit expects to get one result, %1$s found! '.$iter->getSql(),
                count($iter)
             )
```
Comment (cedric-anne) [inline inc/commondbtm.class.php:352]: Maybe using `trigger_error()` could help to have a better trace.
```php
trigger_error(
   sprintf(
      'getFromDBByCrit expects to get one result, %1$s found in query "%2$s".',
      count($iter),
      $iter->getSql()
   ),
   E_USER_WARNING
);
```

Comment (trasher) [reply inc/commondbtm.class.php:352]: This will change the behaviour; and may break existing cases

Comment (cedric-anne) [reply inc/commondbtm.class.php:352]: It will not be blocking, so it will not change the behaviour.

Comment (trasher) [reply inc/commondbtm.class.php:352]: OK, as we've discussed

**Review by orthagh — APPROVED** [review]

**Comment by trasher** [comment]
Tests are failing

**Comment by cedric-anne** [comment]
Hi,

I am closing this PR because you did not follow up on our comments.

Feel free to reopen it if you want to finalize it.

Regards

**Pull request closed without merging**